### DISCUSSION TITLE
Searcher automatically skips specific queries

### DISCUSSION THREAD

**Discussion by namespace-Pt** [castorini/anserini#1714] [Q&A] [answered]
When using BM25 searcher on MSMARCO, I found it automatically skips (retrieves no document) some special queries:
1. containing only one word, e.g. `hopefullymeaning`
2. words are joined by special punctuations instead of space, e.g. `define:semordnilap`

How to deal with this? (Currently I manually modified these queries so that they can be processed)

**Comment by lintool** [comment, marked as answer]
hi @namespace-Pt - The tokenization is determined by Lucene analyzer used during indexing (which has to match the one used during retrieval). You'll need to modify the analyzer if you want to deal with special cases.

In general, in any large, real-world dataset, there are going to be queries/documents that are mangled/mis-tokenized/etc. It's usually not worth the effort to try to fix everything...

**Reply by namespace-Pt** [reply]
Thank you lintool.
